### PR TITLE
fix: added essential CSP directives to public asset distribution

### PR DESCRIPTION
**Pull request by chejimmy**
# Description

Added essential CSP directives to public asset distribution to help mitigate many common web attacks, mostly related to cross-site scripting (XSS).

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Deployed both CDK and cfn template and validated signin and dashboard workflows

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: cdk/lib/endpoints/aws-endpoints.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Endpoint from: https://docs.aws.amazon.com/general/latest/gr/cognito_identity.html
```
Comment (tracy-french) [inline cdk/lib/endpoints/aws-endpoints.ts:1]: Thanks for these comments!

File: cdk/lib/csp/public-asset-directives.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+export function getFontSrcDirective(): string {
+  return "font-src 'self' data:;";
+}
+
+export function getImgSrcDirective(): string {
+  return "img-src 'self' data:;";
+}
+
+export function getScriptSrcDirective(): string {
+  // FIXME: 'unsafe-eval' source to be removed after removal of NPM dependency box-intersect
```
Comment (tracy-french) [inline cdk/lib/csp/public-asset-directives.ts:44]: Nit: Consider making this comment more specific (to IoT App Kit dashboard) and/or linking a GitHub issue.

**Review by tracy-french — APPROVED** [review]
Awesome work as always @chejimmy!

File: cdk/lib/asset/public-asset-stack.ts [hunk 3]
```diff
@@ -108,11 +110,28 @@ export class PublicAssetStack extends Stack {
...
+      this,
+      'PublicAssetHeaders',
+      {
+        securityHeadersBehavior: {
+          contentSecurityPolicy: {
+            contentSecurityPolicy: getPublicAssetCsp(
+              coreServiceUrl,
+              this.region,
+            ),
+            override: true,
```
[the remark this answers is not in the document]
Comment (chejimmy) [reply cdk/lib/asset/public-asset-stack.ts:123]: lemme do a follow up for the nits 👍 
🙏 for the review 🙂

**Pull request merged**